librocksdb-sys/c_api_extensions/write_buffer_manager.cc

```diff
@@ -0,0 +1,37 @@
+// Implementation of `WriteBufferManager` types in `c.h`.
+#include "c_api_extensions/ctypes.hpp"
+
+using namespace ROCKSDB_NAMESPACE;
+
+using std::shared_ptr;
+
+extern "C" {
+void rocksdb_options_set_write_buffer_manager(rocksdb_options_t* opt, rocksdb_write_buffer_manager_t* manager) {
+  opt->rep.write_buffer_manager = manager->rep;
+}
+rocksdb_write_buffer_manager_t* rocksdb_write_buffer_manager_create(size_t buffer_size, bool allow_stall) {
+  auto manager = new rocksdb_write_buffer_manager_t;
+  manager->rep.reset(new WriteBufferManager(buffer_size, {}, allow_stall));
```

Comment:
I prefer `manager->rep = make_shared(buffer_size, {}, allow_stall)` as it allows you to avoid using the `new` keyword, but it's up to you.

Reply:
This is directly copied from the rocksdb code so ill leave it as is